File: bachelor-thesis-odms/MongoDB/JavaODMs/odmstest/src/main/java/cz/cuni/mff/mongodb_java/morphia/models/tpc_h_embedded/OrdersE.java
package cz.cuni.mff.mongodb_java.morphia.models.tpc_h_embedded;

import dev.morphia.annotations.*;

import java.time.LocalDate;

@Embedded
/*@Indexes({
        @Index(fields = @Field(value = "o_orderkey")),
        @Index(fields = @Field(value = "o_custkey"))
})*/
public class OrdersE {
    @Property
    private int o_orderkey;
    @Property
    private int o_custkey;
    private String o_orderstatus;
    public double o_totalprice;
    private LocalDate o_orderdate;
    private String o_orderpriority;
    private String o_clerk;
    private String o_shippriority;
    private String o_comment;
    //private ArrayList<> lineitems;TODO: SOME REFERENCE

    public int get_o_custkey() {
        return o_custkey;
    }

    public OrdersE() {}

    public OrdersE(int o_orderkey, int o_custkey, String o_orderstatus, double o_totalprice, LocalDate o_orderdate, String o_orderpriority, String o_clerk, String o_shippriority, String o_comment) {
        this.o_orderkey = o_orderkey;
        this.o_custkey = o_custkey;
        this.o_orderstatus = o_orderstatus;
        this.o_orderdate = o_orderdate;
        this.o_orderpriority = o_orderpriority;
        this.o_clerk = o_clerk;
        this.o_shippriority = o_shippriority;
        this.o_totalprice = o_totalprice;
        this.o_comment = o_comment;
    }
}

File: evaluation/common-csharp/Models/TPC_H/E/OrdersE.cs
using MongoDB.Entities;

namespace CommonCSharp.Models.TPC_H
{
    public class OrdersE
    {
        public int o_orderkey { get; set; }
        public int o_custkey { get; set; }
        public string o_orderstatus;
        public string o_totalprice;
        public Date o_orderdate;
        public string o_orderpriority;
        public string o_clerk;
        public string o_shippriority;
        public string o_comment;

        public OrdersE(string[] row) : this(
            Convert.ToInt32(row[0]), Convert.ToInt32(row[1]),
            row[2], row[3], new Date(DateTime.Parse(row[4])),
            row[5], row[6], row[7], row[8]) { }

        public OrdersE(int o_orderkey, int o_custkey, string o_orderstatus, string o_totalprice,
            Date o_orderdate, string o_orderpriority, string o_clerk, string o_shippriority, string o_comment)
        {
            this.o_orderkey = o_orderkey;
            this.o_custkey = o_custkey;
            this.o_orderstatus = o_orderstatus;
            this.o_totalprice = o_totalprice;
            this.o_orderdate = o_orderdate;
            this.o_orderpriority = o_orderpriority;
            this.o_clerk = o_clerk;
            this.o_shippriority = o_shippriority;
            this.o_comment = o_comment;
        }
    }
}
